packages/common/src/util/http.js

```diff
@@ -256,15 +256,24 @@ function encodeRequestBody(body) {
 }
 
 async function readResponseBody(response, parseAs) {
-  const contentLength = parseInt(
-    response.headers['content-length'] ?? response.body?.readableLength
-  );
+  let contentLength = -1;
+  if ('content-length' in response.headers) {
+    contentLength = parseInt(response.headers['content-length']);
+  }
+
+  // Try to work out if the response is empty
+  // If so, we should just return without trying to parse it
+  // (turns out that this isn't easy to do!)
+  if (
+    contentLength == 0 ||
+    response.statusCode === 204 ||
```

Comment:
Not at all happy about this stuff. I can't find a really good, robust way to determine if there's any body content in the response. Which seems crazy right?

Reply:
Ah, a good note from undici:

```
Recommended way to ensure there is a body to deserialize is to check if status code is not 204, and `content-type` header starts with `application/json`.
```

https://github.com/nodejs/undici/pull/1516/files